tabular/src/autogluon/tabular/trainer/model_presets/presets.py

```diff
@@ -112,121 +110,168 @@
-                model[AG_ARGS] = dict()
-            if 'model_type' not in model[AG_ARGS]:
-                model[AG_ARGS]['model_type'] = model_type
-            model_type_real = model[AG_ARGS]['model_type']
-            if not inspect.isclass(model_type_real):
-                model_type_real = MODEL_TYPES[model_type_real]
-            default_ag_args = model_type_real._get_default_ag_args()
-            if ag_args is not None:
-                model_extra_ag_args = ag_args.copy()
-                model_extra_ag_args.update(model[AG_ARGS])
-                model[AG_ARGS] = model_extra_ag_args
-            if ag_args_ensemble is not None:
-                model_extra_ag_args_ensemble = ag_args_ensemble.copy()
-                model_extra_ag_args_ensemble.update(model.get(AG_ARGS_ENSEMBLE, dict()))
-                model[AG_ARGS_ENSEMBLE] = model_extra_ag_args_ensemble
-            if ag_args_fit is not None:
-                if AG_ARGS_FIT not in model:
-                    model[AG_ARGS_FIT] = dict()
-                model_extra_ag_args_fit = ag_args_fit.copy()
-                model_extra_ag_args_fit.update(model[AG_ARGS_FIT])
-                model[AG_ARGS_FIT] = model_extra_ag_args_fit
-            if default_ag_args is not None:
-                default_ag_args.update(model[AG_ARGS])
-                model[AG_ARGS] = default_ag_args
-            model_priority = model[AG_ARGS].get('priority', default_priorities.get(model_type, DEFAULT_CUSTOM_MODEL_PRIORITY))
+        models_of_type = hp_level[model_type]
+        if not isinstance(models_of_type, list):
+            models_of_type = [models_of_type]
+        for model in models_of_type:
+            model = clean_model_config(model=model, model_type=model_type, ag_args=ag_args, ag_args_ensemble=ag_args_ensemble, ag_args_fit=ag_args_fit)
+            model[AG_ARGS]['priority'] = model[AG_ARGS].get('priority', default_priorities.get(model_type, DEFAULT_CUSTOM_MODEL_PRIORITY))
+            model_priority = model[AG_ARGS]['priority']
             # Check if model is valid
-            if hyperparameter_tune_kwargs and model[AG_ARGS].get('disable_in_hpo', False):
-                continue  # Not valid
-            elif not model[AG_ARGS].get('valid_stacker', True) and level > 0:
-                continue  # Not valid as a stacker model
-            elif not model[AG_ARGS].get('valid_base', True) and level == 0:
-                continue  # Not valid as a base model
-            priority_dict[model_priority].append(model)
+            is_valid = is_model_valid(model, level=level)
+            if is_valid:
+                priority_dict[model_priority].append(model)
+
     model_priority_list = [model for priority in sorted(priority_dict.keys(), reverse=True) for model in priority_dict[priority]]
-    model_names_set = set()
-    if invalid_model_names is not None:
-        model_names_set.update(invalid_model_names)
+
+    if not silent:
+        logger.log(20, 'Model configs to train:')
```

Comment:
may be more clear: "to train" -> "that will be trained". 
Also should this this info be dumped by default, or should it have slightly lower logging-level priority so it's only output under higher verbosity?

Reply:
Updated. This is controlled by silent argument, which is toggled off if verbosity > 2 in Trainer.